controllers/platform/platform_detection.go

```diff
@@ -33,8 +33,9 @@ var (
 // platform is used to get the PlatformType of the running cluster in a thread-safe manner
 // It is a singleton which is initialized exactly once via Detect() function call.
 type platform struct {
-	isOpenShift bool
-	platform    configv1.PlatformType
+	isOpenShift    bool
+	platform       configv1.PlatformType
+	infrastructure *configv1.Infrastructure
```

Comment:
Instead of storing the entire struct and checking if infrastructure is ROSA every time, can we just have `isROSA bool` and initialize it once?

Reply:
@umangachapagain made the changes by adding `isROSAHCP` bool

Reply:
@umangachapagain have opened an issue since this needs to be discussed https://github.com/red-hat-storage/ocs-operator/issues/2521